### PR TITLE
Updated no-translation-outside-of-function to account for FunctionDeclaration

### PR DESCRIPTION
**Pull request by rassanw**
@zackargyle is this what you had in mine? I don't really know how to test this, though

### CODE REVIEW

File: lib/rules/no-translation-outside-of-function.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ module.exports = function (context) {
             functionCounter--;
         },
 
+        "FunctionDeclaration": function() {
+            functionCounter++;
+        },
+
+        "FunctionDeclaration": function() {
```
Comment (zackargyle) [inline lib/rules/no-translation-outside-of-function.js]: Make this `FunctionDeclaration:exit` and I think we're good.

**Comment by zackargyle** [comment]
If you want to test, you can add this code into your local node_modules code and run eslint locally. Probably the easiest way to test.

**Comment by rassanw** [comment]
Made suggested change!

**Comment by zackargyle** [comment]
Thanks!

**Pull request merged**